docs/auto-instrumentation-install.md

```diff
@@ -0,0 +1,278 @@
+- OpentTelemetry leverages Kubernetes to implement auto-instrumentation in such a way that your applications can be
+  instrumented not by updating the application source code but rather by utilizing Kubernetes resources.
+- Supports a subset of instrumentation libraries compared to manual instrumentation
+
+Manual Instrumentation
+
+- Takes more setup effort compared to auto-instrumentation but offers more customizability
+- Requires editing application source code to include pre-built OpenTelemetry instrumentation libraries
+- Supports more instrumentation libraries and customization of the exported telemetry data compared to
+  auto-instrumentation
+
+In particular, auto-instrumentation is useful for applications that use widely popular frameworks and libraries, as
+these frameworks often have pre-built instrumentation capabilities already available.
+
+## Getting started with auto-instrumentation
+
+### 1. If the cert-manager available in the cluster, deploy it
+
+The cert-manager adds certificates and certificate issuers as resource types in Kubernetes clusters, and simplifies the
+process of obtaining, renewing and using those certificates. You can use the following make commands to deploy the
+cert-manager.
+
+```
+make cert-manager
+
+# If make is not availabe, you can use these commands.
+kubectl get pods -l app=cert-manager --all-namespaces
+kubectl apply -f https://github.com/cert-manager/cert-manager/releases/download/v1.10.0/cert-manager.yaml
+```
+
+### 2. Deploy the Helm Chart with the Operator enabled
+
+You can pass `--set operator.enabled=true` when deploying the chart to enable the operator. You can also use commands
+like in the following example.
+
+```
+VALUES_FILE=examples/enable-operator-and-auto-instrumentation/enable-operator-and-auto-instrumentation-values.yaml
+helm install splunk-otel-collector \
+-f $VALUES_FILE \
+-n monitoring \
+helm-charts/splunk-otel-collector
+```
+
+### 3. Deploy the opentelemetry.io/v1alpha1 Instrumentation
+
+This Instrumentation object is a spec to configure what instrumentation libraries to use for instrumentation. An
+Instrumentation object must be available to the target pod for auto-instrumentation to function. Here is an example
+of a Instrumentation yaml file and how to install it.
+
+<details open>
+<summary>Expand for sample splunk-instrumentation.yaml</summary>
+
+```yaml
+# splunk-instrumentation.yaml
+apiVersion: opentelemetry.io/v1alpha1
+kind: Instrumentation
+metadata:
+  name: splunk-instrumentation
+spec:
+  exporter:
+    endpoint: http://$(SPLUNK_OTEL_AGENT):4317
+  propagators:
+    - tracecontext
+    - baggage
+    - b3
+  dotnet:
+    env:
+      - name: SPLUNK_OTEL_AGENT
+        valueFrom:
+          fieldRef:
+            apiVersion: v1
+            fieldPath: status.hostIP
+      - name: OTEL_EXPORTER_OTLP_ENDPOINT
+        value: http://$(SPLUNK_OTEL_AGENT):4317
+      - name: OTEL_TRACES_EXPORTER
+        value: otlp
```

Comment:
Is there a way to install it automatically from our chart?

Reply:
I spent a large chunk of time this week trying to solve this exact issue, I decided to put this code out this way at the start for now. Their are technical limitations we have to work around, I also think their may be a bug or undocumented functionality (still confirming, reported by other Splunkers as well) that relates to https://github.com/open-telemetry/opentelemetry-operator/issues/886 that has automating this more difficult.

I'm not sure if we should consider this a blocker for initial release. I think we should highly prioritize addressing this, but it is likely more appropriate to do in a follow up PR. Any thoughts?

Reply:
A follow up sounds good to me 👍